src/api/async_element.rs

```diff
@@ -352,6 +352,44 @@ mod tests {
         assert_eq!(router_output.len(), 2001);
     }
 
+    #[test]
+    fn one_async_element_with_empty_stream() {
+        let default_channel_size = 10;
+        let packets = vec![];
+        let packet_generator = immediate_stream(packets.clone());
+
+        let elem0 = AsyncIdentityElement { id: 0 };
+
+        let elem0_link = AsyncElementLink::new(Box::new(packet_generator), elem0, default_channel_size);
+
+        let (s, r) = crossbeam_channel::unbounded();
+        let elem0_drain = elem0_link.consumer;
+        let elem0_collector = ExhaustiveCollector::new(0, Box::new(elem0_link.provider), s);
+        let elem0_overseer = elem0_link.overseer;
+
+        tokio::run(lazy ( || {
+            tokio::spawn(elem0_drain);
+            tokio::spawn(elem0_collector);
+            tokio::spawn(elem0_overseer);
+            Ok(())
+        }));
+
+        let router_output: Vec<_> = r.iter().collect();
+        assert_eq!(router_output, packets);
+    }
+
+    #[test]
+    #[should_panic]
+    fn one_async_element_with_empty_channel() {
```

Comment:
Not sure if this test is complete yet. What is the empty channel here we're testing?

Reply:
we are testing that if the `default_channel_size` is set to 0, it panics.